src-ghc/Pact/Server/PactService.hs

```diff
@@ -91,18 +93,67 @@ runPayload c@Command{..} = do
         _ -> return $ jsonResult mode (cmdToRequestKey c) $ CommandError "Private" Nothing
     Nothing -> runRpc _pPayload
 
-
-
 applyExec :: RequestKey -> ExecMsg ParsedCode -> Command a -> CommandM p CommandResult
 applyExec rk (ExecMsg parsedCode edata) Command{..} = do
   CommandEnv {..} <- ask
   when (null (_pcExps parsedCode)) $ throwCmdEx "No expressions found"
-  (CommandState refStore) <- liftIO $ readMVar _ceState
-  let evalEnv = setupEvalEnv _ceDbEnv _ceEntity _ceMode
-                (MsgData (userSigsToPactKeySet _cmdSigs) edata Nothing _cmdHash) refStore
-  pr <- liftIO $ evalExec evalEnv parsedCode
-  void $ liftIO $ swapMVar _ceState $ CommandState (erRefStore pr)
-  return $ jsonResult _ceMode rk $ CommandSuccess (last (erOutput pr))
-
-applyContinuation :: ContMsg -> [UserSig] -> CommandM p CommandResult
-applyContinuation _ _ = throwCmdEx "Continuation not supported"
+  CommandState {..} <- liftIO $ readMVar _ceState
+  let sigs = userSigsToPactKeySet _cmdSigs
+      evalEnv = setupEvalEnv _ceDbEnv _ceEntity _ceMode
+                (MsgData sigs edata Nothing _cmdHash) _csRefStore
+  EvalResult{..} <- liftIO $ evalExec evalEnv parsedCode
+  newPact <- join <$> mapM (handleYield erInput _cmdSigs) erExec
+  let newState = CommandState erRefStore $ case newPact of
+        Nothing -> _csPacts
+        Just (cp@CommandPact{..}) -> M.insert _cpTxId cp _csPacts
+  void $ liftIO $ swapMVar _ceState newState
+  return $ jsonResult _ceMode rk $ CommandSuccess (last erOutput)
+
+-- Better name?
+handleYield :: [Term Name] -> [UserSig] -> PactExec -> CommandM p (Maybe CommandPact)
+handleYield em cmdSigs PactExec{..} = do
+  CommandEnv{..} <- ask
+  --TODO: handle entity?
```

Comment:
We'd need some way to fake multiple entities, which are different servers with different databases. This is why, in fact, there was never pacts in the dev server until now